### PR TITLE
Handle multiple substitutions in custom todo keyword

### LINKED ISSUE #299: Todo keywords with multiple hyphens break keyword cycling
### Describe the bug

When adding a custom Todo keyword that contains at least two hyphens, the cycling between keywords with `cit` or `ciT` does not remove the keyword in question.

### Steps to reproduce

1. `echo "* foo" >> foo.org`
2.  `vim -u minimal_init.lua foo.org`
3.   `cit` three times

### Expected behavior

The cycle should work just like with non-hyphen or one-hyphen keywords

### Minimal init.lua

```
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]

vim.keymap.set("n", "<Space>", "<NOP>", {noremap = true, silent = true})
vim.g.mapleader = ' '

local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'

local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {'nvim-treesitter/nvim-treesitter', run = ':TSUpdate'},
      {'kristijanhusak/orgmode.nvim', branch = 'master' },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
    },
  }
end

_G.load_config = function()
  require('orgmode').setup_ts_grammar()
  require('nvim-treesitter.configs').setup({})

  vim.cmd[[packadd nvim-treesitter]]
  vim.cmd[[TSUpdate]]

  require('orgmode').setup({
    org_todo_keywords = { "TODO", "THREE-WORD-KEYWORD", "|", "DONE" },
  })
end

if vim.fn.isdirectory(install_path) == 0 then
  vim.fn.system { 'git', 'clone', 'https://github.com/wbthomason/packer.nvim', install_path }
  load_plugins()
  require('packer').sync()
  vim.cmd [[autocmd User PackerComplete ++once lua load_config()]]
else
  load_plugins()
  require('packer').sync()
  _G.load_config()
end
```
### Screenshots and recordings

multi-hyphens

### OS / Distro

Ubuntu 20.04

### Neovim version/commit

0.7.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jgollenz**
fixes #299

### CODE REVIEW

File: lua/orgmode/treesitter/headline.lua [hunk 1]
```diff
@@ -124,7 +124,9 @@ function Headline:todo()
 
   local text = query.get_node_text(todo_node, 0)
   for _, word in ipairs(keywords) do
-    local todo = text:match(word:gsub('-', '%%-'))
+    -- there may be multiple substitutions necessary
+    subsituted_word = word:gsub('-', '%%-')
```
[the remark this answers is not in the document]
Comment (jgollenz) [reply lua/orgmode/treesitter/headline.lua]: Absolutely. Done

**Comment by jgollenz** [comment]
Please hold on merging this. I just found a bug: 

`[orgmode] /usr/share/nvim/runtime/lua/vim/treesitter/query.lua:185: attempt to index local 'node' (a nil value) `

When going from TODO type keywords to Done type keywords IF the headline has a timestamp. Need to further investigate

**Comment by jgollenz** [comment]
@kristijanhusak it *looks* like this PR is not rleated to #309. Should be good to merge

**Comment by kristijanhusak** [comment]
Thanks!

**Pull request merged**